<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1088</width>
    <height>801</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>NN Maker</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <widget class="QChartWidget" name="chart" native="true">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>10</y>
      <width>841</width>
      <height>761</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>621</width>
      <height>561</height>
     </size>
    </property>
    <property name="cursor">
     <cursorShape>OpenHandCursor</cursorShape>
    </property>
    <property name="toolTip">
     <string>How much data to generate</string>
    </property>
    <property name="toolTipDuration">
     <number>2</number>
    </property>
    <property name="whatsThis">
     <string>Test</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="scrollArea">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>241</width>
      <height>771</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>241</width>
      <height>600</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>241</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents_3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>241</width>
       <height>771</height>
      </rect>
     </property>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>620</y>
        <width>221</width>
        <height>3</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>221</width>
        <height>3</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>3</height>
       </size>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>221</width>
        <height>451</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>221</width>
        <height>451</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>221</width>
        <height>451</height>
       </size>
      </property>
      <property name="font">
       <font>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="title">
       <string>Network configuration</string>
      </property>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <italic>true</italic>
         <bold>false</bold>
         <kerning>true</kerning>
        </font>
       </property>
       <property name="text">
        <string>Nb. neurons :</string>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
      </widget>
      <widget class="QSpinBox" name="neurons">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>30</y>
         <width>91</width>
         <height>28</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>999</number>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>60</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <italic>true</italic>
         <bold>false</bold>
         <kerning>true</kerning>
        </font>
       </property>
       <property name="text">
        <string>Activation :</string>
       </property>
      </widget>
      <widget class="QComboBox" name="activ">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>60</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Sigmoid</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RELU</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>ELU</string>
        </property>
       </item>
      </widget>
      <widget class="QPushButton" name="addLayer">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>201</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Add Layer</string>
       </property>
      </widget>
      <widget class="QListLayerWidget" name="listWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>140</y>
         <width>201</width>
         <height>171</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>320</y>
         <width>91</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <italic>true</italic>
         <bold>false</bold>
         <kerning>true</kerning>
        </font>
       </property>
       <property name="text">
        <string>Nb. features :</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="features">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>320</y>
         <width>91</width>
         <height>28</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>Number of entries in your network</string>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::UpDownArrows</enum>
       </property>
       <property name="keyboardTracking">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
      </widget>
      <widget class="QPushButton" name="clear">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>410</y>
         <width>71</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Clear</string>
       </property>
      </widget>
      <widget class="QComboBox" name="loss">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>370</y>
         <width>201</width>
         <height>31</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Binary Cross Entropy</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>350</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <weight>50</weight>
         <italic>true</italic>
         <bold>false</bold>
         <kerning>true</kerning>
        </font>
       </property>
       <property name="text">
        <string> Loss function :</string>
       </property>
      </widget>
      <widget class="QProgressBar" name="pbar_build">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>410</y>
         <width>121</width>
         <height>31</height>
        </rect>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QPushButton" name="build">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>410</y>
         <width>121</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Build</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
      <zorder>pbar_build</zorder>
      <zorder>label_2</zorder>
      <zorder>neurons</zorder>
      <zorder>label</zorder>
      <zorder>activ</zorder>
      <zorder>addLayer</zorder>
      <zorder>listWidget</zorder>
      <zorder>label_3</zorder>
      <zorder>features</zorder>
      <zorder>clear</zorder>
      <zorder>loss</zorder>
      <zorder>label_4</zorder>
      <zorder>build</zorder>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>630</y>
        <width>221</width>
        <height>141</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>221</width>
        <height>141</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>221</width>
        <height>141</height>
       </size>
      </property>
      <property name="title">
       <string>Training</string>
      </property>
      <widget class="QDoubleSpinBox" name="learningRate">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>60</y>
         <width>91</width>
         <height>28</height>
        </rect>
       </property>
       <property name="locale">
        <locale language="English" country="Europe"/>
       </property>
       <property name="correctionMode">
        <enum>QAbstractSpinBox::CorrectToPreviousValue</enum>
       </property>
       <property name="decimals">
        <number>10</number>
       </property>
       <property name="maximum">
        <double>999.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="stepType">
        <enum>QAbstractSpinBox::DefaultStepType</enum>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>30</y>
         <width>66</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <italic>true</italic>
        </font>
       </property>
       <property name="text">
        <string>Epochs :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <italic>true</italic>
        </font>
       </property>
       <property name="text">
        <string>Learning rate :</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="epochs">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>30</y>
         <width>91</width>
         <height>28</height>
        </rect>
       </property>
       <property name="minimum">
        <number>100</number>
       </property>
       <property name="maximum">
        <number>10000000</number>
       </property>
       <property name="singleStep">
        <number>100</number>
       </property>
       <property name="value">
        <number>1000</number>
       </property>
      </widget>
      <widget class="QPushButton" name="train">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>201</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Train</string>
       </property>
      </widget>
      <widget class="QProgressBar" name="pbar_train">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>201</width>
         <height>31</height>
        </rect>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <zorder>learningRate</zorder>
      <zorder>label_7</zorder>
      <zorder>label_8</zorder>
      <zorder>epochs</zorder>
      <zorder>pbar_train</zorder>
      <zorder>train</zorder>
     </widget>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>460</y>
        <width>221</width>
        <height>3</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>221</width>
        <height>3</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>3</height>
       </size>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>470</y>
        <width>221</width>
        <height>141</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>221</width>
        <height>141</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>221</width>
        <height>141</height>
       </size>
      </property>
      <property name="title">
       <string>Data</string>
      </property>
      <widget class="QTabWidget" name="tabWidget">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>20</y>
         <width>221</width>
         <height>121</height>
        </rect>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_3">
        <attribute name="title">
         <string>Generate</string>
        </attribute>
        <widget class="QRadioButton" name="linear_data">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>0</y>
           <width>111</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>Linear</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QRadioButton" name="balanced_data">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>0</y>
           <width>111</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>Balanced</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="circle_data">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>111</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>Circle</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="class3_data">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>20</y>
           <width>111</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>3 Class</string>
         </property>
        </widget>
        <widget class="QPushButton" name="data_details_2">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>50</y>
           <width>101</width>
           <height>27</height>
          </rect>
         </property>
         <property name="text">
          <string>See details</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="nb_data">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>61</width>
           <height>28</height>
          </rect>
         </property>
         <property name="toolTip">
          <string>Number of data to generate</string>
         </property>
         <property name="toolTipDuration">
          <number>-1</number>
         </property>
         <property name="minimum">
          <number>100</number>
         </property>
         <property name="maximum">
          <number>5000</number>
         </property>
         <property name="singleStep">
          <number>100</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_4">
        <attribute name="title">
         <string>Load</string>
        </attribute>
        <widget class="QPushButton" name="browse">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>91</width>
           <height>27</height>
          </rect>
         </property>
         <property name="text">
          <string>Browse..</string>
         </property>
        </widget>
        <widget class="QPushButton" name="data_details">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>50</y>
           <width>101</width>
           <height>27</height>
          </rect>
         </property>
         <property name="text">
          <string>See details</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_19">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>31</width>
           <height>19</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>file:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="choosed_file">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>16</y>
           <width>161</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>[Not selected]</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </widget>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1088</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuProject">
    <property name="title">
     <string>Project</string>
    </property>
    <addaction name="load"/>
    <addaction name="save"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuProject"/>
  </widget>
  <action name="nouveau">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="save">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="load">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="about">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="exit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartWidget</class>
   <extends>QWidget</extends>
   <header>QChartWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QListLayerWidget</class>
   <extends>QListWidget</extends>
   <header>QListLayerWidget.hpp</header>
   <slots>
    <slot>slotAddLayerBtnClicked(int,QString)</slot>
    <slot>slotClearBtnClicked()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
